### PR TITLE
Improved Symbol Search / Goto Definition

### PR DESCRIPTION
**Pull request by joeandaverde**
Still pending some testing but this should provide improved experience navigating Elm files.

### CODE REVIEW

File: src/elmWorkspaceSymbols.ts [hunk 1]
```diff
@@ -1,99 +1,96 @@
...
+    if (matchedFiles.length === 1) {
+      await this.indexDocument(await vscode.workspace.openTextDocument(matchedFiles[0]));
+    }
+  }
+
+  private async indexDocument(document: TextDocument) {
+    const updatedSymbols = await processDocument(document);
+
+    updatedSymbols.forEach(s => {
+      this.symbolsByContainer[s.containerName] = (this.symbolsByContainer[s.containerName] || []).concat(s);
```
Comment (joeandaverde) [inline src/elmWorkspaceSymbols.ts]: This needs to properly remove stale index entries.

File: src/elmDelphi.ts [hunk 2]
```diff
@@ -50,21 +50,26 @@ function readFile(filePath: string) {
...
-  let myArray;
-  while ((myArray = regex.exec(elmCode)) !== null) {
-    const moduleName = myArray[1];
-    const alias = myArray[2];
-    const exposing = myArray[3];
-    imports.push({
-      moduleName: moduleName,
-      alias: alias || moduleName,
-      exposed: exposing === undefined ? null : exposing.split(',').map(str => str.trim()),
+  return elmCode
```
Comment (joeandaverde) [inline src/elmDelphi.ts]: This parse function is no longer needed.

File: src/elmWorkspaceSymbols.ts [hunk 3]
```diff
@@ -1,99 +1,96 @@
 import * as vscode from 'vscode';
 
-import { TextDocument } from 'vscode';
+import { TextDocument, SymbolInformation } from 'vscode';
 import { processDocument } from './elmSymbol';
 
 const config = vscode.workspace.getConfiguration('elm');
 
 export class ElmWorkspaceSymbolProvider
```
Comment (joeandaverde) [inline src/elmWorkspaceSymbols.ts]: This provider could expose functionality to get parsed modules for use in other places e.g. elmDelphi.

**Comment by Krzysztof-Cieslak** [comment]
Hey, can you please rebase the PR and resolve conflicts. Thanks a lot.

**Comment by joeandaverde** [comment]
This should be ready to merge. Fixed the symbol tests too. 

Please verify. I've done quite a bit of testing in the elm module parser and using my > 50K line elm project.

**Comment by Krzysztof-Cieslak** [comment]
Thanks a lot, looks good to me, but I'll play with it locally more before releasing new version of the plugin.

**Pull request merged**